src/moments.jl

```diff
@@ -161,87 +161,57 @@ end
 
 ##### General central moment
 function _moment2(v::AbstractArray{<:Real}, m::Real; corrected=false)
-    n = length(v)
-    s = 0.0
-    for i = 1:n
-        @inbounds z = v[i] - m
-        s += z * z
-    end
-    varcorrection(n, corrected) * s
+    s = sum(x->abs2(x-m), v, init=zero(m))
+    return varcorrection(length(v), corrected) * s
 end
 
 function _moment2(v::AbstractArray{<:Real}, wv::AbstractWeights, m::Real; corrected=false)
-    n = length(v)
-    s = 0.0
-    for i = 1:n
-        @inbounds z = v[i] - m
-        @inbounds s += (z * z) * wv[i]
-    end
-
-    varcorrection(wv, corrected) * s
+    s = sum(i -> (@inbounds abs2(v[i] - m) * wv[i]), eachindex(v), init=zero(m))
+    return varcorrection(wv, corrected) * s
 end
 
 function _moment3(v::AbstractArray{<:Real}, m::Real)
-    n = length(v)
-    s = 0.0
-    for i = 1:n
-        @inbounds z = v[i] - m
-        s += z * z * z
-    end
-    s / n
+    s = sum(x->(x-m)^3, v, init=zero(m))
+    return s/length(v)
 end
 
 function _moment3(v::AbstractArray{<:Real}, wv::AbstractWeights, m::Real)
-    n = length(v)
-    s = 0.0
-    for i = 1:n
-        @inbounds z = v[i] - m
-        @inbounds s += (z * z * z) * wv[i]
-    end
-    s / sum(wv)
+    s = sum(
+        i -> (@inbounds (z = (v[i] - m); z * z * z * wv[i])),
+        eachindex(v),
+        init=zero(m),
+    )
```

Comment:
Same comments as above:
```suggestion
    init = (zero(eltype(v)) - zero(m))^3 * zero(eltype(wv))
    s = sum(
        i -> (v[i] - m)^3 * wv[i],
        eachindex(v, wv);
        init=init,
    )
```